source/common/init/manager_impl.cc

```diff
@@ -51,18 +54,34 @@ void ManagerImpl::initialize(const Watcher& watcher) {
 
     // Attempt to initialize each target. If a target is unavailable, treat it as though it
     // completed immediately.
+    uint32_t target_index = 0;
     for (const auto& target_handle : target_handles_) {
       if (!target_handle->initialize(watcher_)) {
-        onTargetReady();
+        onTargetReady(target_names_[target_index]);
       }
+      target_index++;
     }
   }
 }
 
-void ManagerImpl::onTargetReady() {
+// For init manager to query unready targets
+void ManagerImpl::checkUnreadyTargets() {
+  ENVOY_LOG(debug, "QUERY: List of unready targets..................");
+  for(const auto &unready_target : target_names_count_) {
+    ENVOY_LOG(debug, "{} x {}", unready_target.first, unready_target.second);
```

Comment:
I was hoping Manager could provide a function to dump to protobuf message and eventually used by admin handler.
It's no harm to log though.